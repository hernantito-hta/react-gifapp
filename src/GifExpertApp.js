import React, { Fragment, useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {
    const [categories, setCategories] = useState(defaultCategories);
    // const handleAddItem = (e) => {
    //     setCategories(categs => [...categs, `Item ${Math.random()}`]);
    // };

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = {setCategories} />
            <hr/>
            {/* <button onClick={handleAddItem}>Add item</button> */}
            <ol>
                {
                    categories.map((item, i) => {
                        return <GifGrid
                            key = {i} 
                            category = {item}
                            />;
                    })
                }
            </ol>

        </Fragment>
    );
};

export default GifExpertApp;