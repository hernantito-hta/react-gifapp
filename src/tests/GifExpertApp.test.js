import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Testing component <GifExpertApp />', () => {
    test('should render correctly', () => {
        let wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should display categories', () => {
        const categories = ['One punch', 'Samurai X']
        let wrapper = shallow(<GifExpertApp  defaultCategories = {categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
