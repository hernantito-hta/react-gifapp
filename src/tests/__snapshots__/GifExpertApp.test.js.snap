// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <GifExpertApp /> should display categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="0"
    />
    <GifGrid
      category="Samurai X"
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Testing component <GifExpertApp /> should render correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
